### PR TITLE
TUNIC: Missing slot data bugfix

### PR DESCRIPTION
**Pull request by silent-destroyer**
## What is this fixing or adding?
Fixing an issue where certain data was not getting added to slot data, causing issues in the client mod.

This was due to the way item creation was being handled for items that change classification based on options; the items were being created but not added to slot data, which the client uses for its in-game hint system. I cleaned up the creation of those items by adding an optional argument to the `create_item` function to allow overriding an items default classification on creation, and instead add the necessary item data to slot data at the end of `create_items`

## How was this tested?
Generating test seeds with the options that were causing issues and successfully connecting with the client mod

### CODE REVIEW

File: worlds/tunic/__init__.py [hunk 1]
```diff
@@ -151,9 +151,13 @@ def stage_generate_early(cls, multiworld: MultiWorld) -> None:
                     if new_cxn:
                         cls.seed_groups[group]["plando"].value.append(cxn)
 
-    def create_item(self, name: str) -> TunicItem:
+    def create_item(self, name: str, classification: ItemClassification = None) -> TunicItem:
         item_data = item_table[name]
-        return TunicItem(name, item_data.classification, self.item_name_to_id[name], self.player)
+        tunic_item = TunicItem(name, classification if classification else item_data.classification,
```
Comment (ScipioWright) [inline worlds/tunic/__init__.py:156]:
```suggestion
        tunic_item = TunicItem(name, classification or item_data.classification,
```
Up to you though o:

**Review by ScipioWright — APPROVED** [review]
Good change since it stops the same mistake from repeating itself in the future

**Review by Exempt-Medic — APPROVED** [review]
Gens just fine, changes LGTM, slot_data before and after are very similar. Had a question about the `or` but it's just a ~~silly~~ neat Python trick I didn't know

**Review by NewSoupVi — APPROVED** [review]

**Comment by NewSoupVi** [comment]
For some reason it won't let me merge this

image

Oh well I'll try again later

**Pull request merged**